devel/docker/Portfile

```diff
@@ -14,51 +13,30 @@ description         The open-source application container engine
 long_description    Docker is an open source project to pack, ship \
                     and run any application as a lightweight container.
 
-depends_build       port:go
+checksums           rmd160 28d0ad65c73555c779aee7b34daa9032d5d98eac \
+                    sha256 8e77a39b372f4b8ffb9540bc8ae9bcff6516c716a54d68043e2b6b84780b0cf3 \
+                    size   7518166
 
-checksums           rmd160  e03f7ceb997f9241d822ee15573ef68e870bb11b \
-                    sha256  14d6e2fb393b4a10ef5d5112f6bd931948349e52cb793b6fcf945a034997c983 \
-                    size    7166797
-
-use_configure       no
-
-use_parallel_build  no
-build.cmd           hack/make.sh
-build.target        dynbinary-client
-build.env           AUTO_GOPATH=1
+build.target        github.com/docker/cli/cmd/docker
 
 pre-build {
     set distfile_dirname [exec tar tzf ${distpath}/[lindex ${distfiles} 0] | sed -n 1p]
     set docker_gitcommit [string map "${github.author}-${github.project}- {} / {}" ${distfile_dirname}]
     if {![regexp -nocase -- {^[0-9a-f]{7}$} ${docker_gitcommit}]} {
         return -code error "can't determine git commit"
     }
-    build.env-append DOCKER_GITCOMMIT=${docker_gitcommit}
-}
-
-destroot {
-    xinstall -d ${destroot}${prefix}/bin
-    xinstall -m 755 ${worksrcpath}/bundles/${version}/${build.target}/${name}-${version} ${destroot}${prefix}/bin/${name}
-}
 
-variant bash_completion {
-    depends_run-append path:etc/bash_completion:bash-completion
+    set val [clock microseconds]
+    set seconds_precision [expr { $val / 1000000 }]
+    set build_time [clock format $seconds_precision -format "%a %b %d %H:%M:%S %Y"]
 
-    post-destroot {
-        set completions_path ${prefix}/share/bash-completion/completions
-        xinstall -d ${destroot}${completions_path}
-        xinstall -m 644 ${worksrcpath}/contrib/completion/bash/${name} ${destroot}${completions_path}/
-    }
+    build.cmd ${go.bin} build -ldflags \
+        "\"-X github.com/docker/cli/cli/version.GitCommit=${docker_gitcommit} -X github.com/docker/cli/cli/version.Version=${version} -X \\\"github.com/docker/cli/cli/version.BuildTime=${build_time}\\\"\""
 }
 
-variant zsh_completion {
-    depends_run-append path:${prefix}/bin/zsh:zsh
-
-    post-destroot {
-        set site-functions ${destroot}${prefix}/share/zsh/site-functions
-        xinstall -d ${site-functions}
-        xinstall ${worksrcpath}/contrib/completion/zsh/_${name} ${site-functions}/
-    }
+destroot {
+    xinstall -d ${destroot}${prefix}/bin
```

Comment:
You don't need to create this or other standard directories; MacPorts does that for you before the destroot phase begins.

Reply:
Fixed.